src/object.rs

```diff
@@ -254,11 +256,17 @@ impl Object {
     /// # Ok(())
     /// # }
     /// ```
-    pub async fn range_read(&self, range: impl RangeBounds<u64>) -> Result<Vec<u8>> {
-        let op = OpRead::new(self.path(), range)?;
+    pub async fn range_read(&self, range: impl RangeBounds<u64> + Clone) -> Result<Vec<u8>> {
```

Comment:
Adding `Clone` for the range is a breaking change. Can we implement without `clone`?